### PR TITLE
Fix #284 : No output when responses fail validation

### LINKED ISSUE #284: No output when responses fail validation
Response documents which fail validation result in a completely blank response to clients. The validation error message also doesn't end up in the normal log, and instead seem to end up in localhost.$DATE.log (at least for Tomcat).

An error should be presented to the user when validation fails, and validation errors should be logged to the main log.

### PR DESCRIPTION
**Pull request by CarstenHollmann**

### CODE REVIEW

**Comment by CarstenHollmann** [comment]
Fixes #284

File: bindings/rest/code/src/main/java/org/n52/sos/binding/rest/RestBinding.java [hunk 1]
```diff
@@ -151,7 +151,17 @@ public void doGetOperation(HttpServletRequest request,
         doOperation(request, response);
     }
 
-     /*
+    @Override
```
Comment (autermann) [inline bindings/rest/code/src/main/java/org/n52/sos/binding/rest/RestBinding.java]: Tabs ;)

File: core/api/src/main/java/org/n52/sos/binding/HandleOwsExceptionReport.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.n52.sos.ogc.ows.OwsExceptionReport;
+
+/**
+ * @author <a href="mailto:c.hollmann@52north.org">Carsten Hollmann</a>
+ * @since 4.4.0
+ *
+ */
+public interface HandleOwsExceptionReport {
+	
+    public abstract Object handleOwsExceptionReport(HttpServletRequest request, HttpServletResponse response,
```
Comment (autermann) [inline core/api/src/main/java/org/n52/sos/binding/HandleOwsExceptionReport.java]: `public abstract` can be removed

File: core/api/src/main/java/org/n52/sos/binding/HandleOwsExceptionReport.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.n52.sos.exception.HTTPException;
+import org.n52.sos.ogc.ows.OwsExceptionReport;
+
+/**
+ * @author <a href="mailto:c.hollmann@52north.org">Carsten Hollmann</a>
+ * @since 4.4.0
+ *
+ */
+public interface HandleOwsExceptionReport {
```
Comment (autermann) [inline core/api/src/main/java/org/n52/sos/binding/HandleOwsExceptionReport.java]: how about `OwsExceptionReportHandler`?

File: core/api/src/main/java/org/n52/sos/encode/AbstractServiceResponseWriter.java [hunk 4]
```diff
@@ -55,33 +55,30 @@
...
+				if (encode != null) {
+					ResponseWriter<Object> writer = ResponseWriterRepository.getInstance().getWriter(encode.getClass());
+					if (writer == null) {
+						throw new RuntimeException("no writer for " + encode.getClass() + " found!");
+					}
+					writer.write(encode, out, responseProxy);
+				}
+			}
+		}
+	}
```
Comment (autermann) [inline core/api/src/main/java/org/n52/sos/encode/AbstractServiceResponseWriter.java]: so many tabs...

File: core/api/src/main/java/org/n52/sos/util/http/HTTPUtils.java [hunk 5]
```diff
@@ -120,13 +122,63 @@ public static InputStream getInputStream(HttpServletRequest req) throws IOExcept
...
     }
 
+    /**
+     * @param request
+     * @param response
+     * @param contentType
+     * @param object
+     * @throws IOException
+     * @throws HTTPException
+     */
```
Comment (autermann) [inline core/api/src/main/java/org/n52/sos/util/http/HTTPUtils.java]: you should really remove these dummy JavaDocs...

**Comment by autermann** [comment]
Waiting for 4 spaces :)

**Pull request merged**